auditbeat/module/audit/file/eventreader_fsevents.go

```diff
@@ -0,0 +1,205 @@
+	fsevents.ItemXattrMod:      "ItemXattrMod",
+	fsevents.ItemIsFile:        "ItemIsFile",
+	fsevents.ItemIsDir:         "ItemIsDir",
+	fsevents.ItemIsSymlink:     "ItemIsSymlink",
+}
+
+// NewEventReader creates a new EventProducer backed by FSEvents macOS facility.
+func NewEventReader(c Config) (EventProducer, error) {
+	stream := &fsevents.EventStream{
+		Paths: c.Paths,
+		// NoDefer: Ignore Latency field and send events as fast as possible.
+		//          Useful as latency has one second granularity.
+		//
+		// WatchRoot: Will send a notification when some element changes along
+		// 			the path being watched (dir moved or deleted).
+		//
+		// FileEvents: Get events for files not just directories
+		Flags: fsevents.NoDefer | fsevents.WatchRoot | fsevents.FileEvents,
+	}
+
+	// IgnoreSelf: Avoid infinite looping when auditbeat writes to a
+	//			   watched directory. If specified tests won't work.
+	if !underTest {
+		stream.Flags |= fsevents.IgnoreSelf
+	}
+
+	inodes := make(map[uint64]bool)
+	if !c.Recursive {
+		for _, path := range c.Paths {
+			if inode, err := getInode(path); err == nil {
+				debugf("using path:%s inode:%v", path, inode)
+				inodes[inode] = true
+			} else {
+				logp.Warn("%v failed to get inode for '%s': %v", logPrefix, path, err)
+			}
+		}
+	}
+	return &fsreader{
+		stream:        stream,
+		config:        c,
+		eventC:        make(chan Event, 1),
+		watchedInodes: inodes,
+	}, nil
+}
+
+func (r *fsreader) Start(done <-chan struct{}) (<-chan Event, error) {
+	r.stream.Start()
+	go r.consumeEvents(done)
+	logp.Info("%v started FSEvents watcher recursive:%v", logPrefix, r.config.Recursive)
+	return r.eventC, nil
+}
+
+func (r *fsreader) consumeEvents(done <-chan struct{}) {
+	defer close(r.eventC)
+	defer r.stream.Stop()
+
+	for {
+		select {
+		case <-done:
+			debugf("Terminated")
+			return
+		case events := <-r.stream.Events:
+			for _, event := range events {
+				if !r.isWatched(event.Path) {
+					debugf("Ignoring FSEvents event: path=%v", event.Path)
+					continue
+				}
+				debugf("Received FSEvents event: id=%d path=%v flags=%s",
+					event.ID, event.Path, flagsToString(event.Flags))
+				start := time.Now()
+				e := NewEvent(event.Path, flagsToAction(event.Flags), SourceFSNotify,
+					r.config.MaxFileSizeBytes, r.config.HashTypes)
+
+				e.rtt = time.Since(start)
+				r.eventC <- e
+			}
+		}
+	}
+}
+
+func flagsToAction(flags fsevents.EventFlags) Action {
+	action := None
+	for flag, act := range flagToAction {
+		if (flags & flag) != 0 {
+			action |= act
+		}
+	}
+	return action
+}
+
+func flagsToString(flags fsevents.EventFlags) string {
+	var list []string
+	for key, name := range flagNames {
+		if 0 != flags&key {
+			list = append(list, name)
+		}
+	}
+	return strings.Join(list, "|")
+}
+
+func getInode(path string) (uint64, error) {
+	info, err := os.Lstat(path)
+	if err != nil {
+		return 0, errors.Wrap(err, fmt.Sprintf("failed to stat"))
+	}
+	meta, err := NewMetadata(path, info)
```

Comment:
Given that you need only the inode and file type I would use the os.FileInfo directly to avoid some of the extra work that NewMetadata does (like lookup username and group).